include/aws/crt/iot/DeviceDefender.h

```diff
@@ -0,0 +1,200 @@
+#pragma once
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+#include <aws/crt/io/EventLoopGroup.h>
+#include <aws/crt/mqtt/MqttClient.h>
+
+#include <aws/iotdevice/device_defender.h>
+
+namespace Aws
+{
+    namespace Crt
+    {
+
+        namespace Io
+        {
+            class EventLoopGroup;
+        }
+
+        namespace Mqtt
+        {
+            class MqttConnection;
+        }
+
+        namespace Iot
+        {
+
+            /**
+             * Invoked upon DeviceDefender V1 task cancellation.
+             */
+            using OnDefenderV1TaskCancelledHandler = std::function<void(void *)>;
+
+            using DeviceDefenderReportFormat = aws_iotdevice_defender_report_format;
+
+            /**
+             * Enum used to expose the status of a DeviceDefenderV1 task.
+             */
+            enum class DeviceDefenderV1ReportTaskStatus
+            {
+                Ready = 0,
+                Running = 1,
+                Stopped = 2,
+                Failed = 3,
+            };
+
+            /**
+             * Represents a unique configuration for a device defender V1 report task. You can use a single instance of
+             * this class PER task you want to create.
+             */
+            class AWS_CRT_CPP_API DeviceDefenderV1ReportTaskConfig final
+            {
+
+                friend class DeviceDefenderV1ReportTask;
+                friend class DeviceDefenderV1ReportTaskConfigBuilder;
+
+              public:
+                /**
+                 * Creates a client configuration for use with making new AWS Iot specific MQTT Connections with MTLS.
+                 */
+                DeviceDefenderV1ReportTaskConfig(
+                    std::shared_ptr<Mqtt::MqttConnection> mqttConnection,
+                    ByteCursor thingName,
+                    Io::EventLoopGroup &eventLoopGroup,
+                    DeviceDefenderReportFormat reportFormat,
+                    uint64_t taskPeriodNs,
+                    uint64_t networkConnectionSamplePeriodNs,
+                    OnDefenderV1TaskCancelledHandler &&onCancelled,
+                    void *cancellationUserdata) noexcept;
+
+                /**
+                 * @return the value of the last aws error encountered by operations on this instance.
+                 */
+                int LastError() const noexcept { return m_lastError; }
+
+              private:
+                std::shared_ptr<Mqtt::MqttConnection> mqttConnection;
+                ByteCursor thingName;
+                Io::EventLoopGroup &eventLoopGroup;
+                DeviceDefenderReportFormat reportFormat;
+                uint64_t taskPeriodNs;
+                uint64_t networkConnectionSamplePeriodNs;
+                OnDefenderV1TaskCancelledHandler onCancelled;
+                void *cancellationUserdata;
```

Comment:
Consider to reuse existing C `struct aws_iotdevice_defender_report_task_config` **if possible**.

Reply:
That makes sense. I've refactored to reuse the existing struct in updated commits.